homeassistant/components/sonos/__init__.py

```diff
@@ -1,10 +1,26 @@
 """Support to embed Sonos."""
-from homeassistant import config_entries
-from homeassistant.helpers import config_entry_flow
+import voluptuous as vol
 
+from homeassistant import config_entries
+from homeassistant.components.media_player import DOMAIN as MP_DOMAIN
+from homeassistant.const import CONF_HOSTS
+from homeassistant.helpers import config_entry_flow, config_validation as cv
 
 DOMAIN = 'sonos'
 
+CONF_ADVERTISE_ADDR = 'advertise_addr'
+CONF_INTERFACE_ADDR = 'interface_addr'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        MP_DOMAIN: vol.Schema({
+            vol.Optional(CONF_ADVERTISE_ADDR): cv.string,
+            vol.Optional(CONF_INTERFACE_ADDR): cv.string,
+            vol.Optional(CONF_HOSTS): vol.All(cv.ensure_list_csv, [cv.string]),
```

Comment:
Why do we need to use `cv.ensure_list_csv` instead of `cv.ensure_list`? Is there some backwards compatibility we need to handle?

Our standard is to only use `cv.ensure_list` for lists of options.

Reply:
The old code did a manual split.

I don’t personally mind another breaking change but I asked about this in Discord and did not get a response so I chose the non-breaking option.

Reply:
Backwards compatibility is the only use case for this validator, so then I think it's fine. :+1: